app/components/styleguides/button.js

```diff
@@ -0,0 +1,82 @@
+import Component from '@ember/component';
+import abstractMethod from 'travis/utils/abstract-method';
+import { computed } from '@ember/object';
+
+const DEFAULT_SHADES = {
+  light: 100,
+  main: 400,
+  dark: 500,
+};
+
+const DEFAULT_COLOR = 'blue';
+const COLORSETS = {
+  disabled: {
+    color: 'grey',
+    shades: {
+      light: 100,
+      main: 300,
+      dark: 300,
+    }
+  }
+};
+
+export const DEFAULT_VARIANT = 'fill';
+export const VARIANTS = {
+  fill: (color, shades) => [
+    `bg-${color}-${shades['main']}`,
+    'text-white',
+    `hover:bg-${color}-${shades['dark']}`,
+  ],
+
+  outline: (color, shades) => [
+    `text-${color}-${shades['main']}`,
+    'border',
+    'border-solid',
+    `border-${color}-${shades['main']}`,
+    `hover:bg-${color}-${shades['light']}`,
+  ],
+};
+
+export default Component.extend({
+  tagName: 'button',
```

Comment:
Ahh that's a good point! It's different enough that, since that seems to be where Ember is headed, maybe it makes sense to take that approach now. And direct binding of classes and attributes might be nice in some ways. I'll try this out